### PR TITLE
Allow to define custom XCom class

### LINKED ISSUE #8059: [AIP-31] Enable arbitrary XCom class
**Description**

- Enable custom XCom backend functionality for custom serialization/deserialization.
- XCom backend can be set from airflow.cfg and defaults to existing `airflow.models.XCom`. 

**Use case / motivation**
This comes from a discussion with @turbaszek . We were talking that it would be nice to have a flexible XCom backend that allowed us to serialize/deserialize XCom values using custom logic. A possible extension here would be to enable large objects to be serialized into a custom file system while XCom would only include a pointer (and metadata) of the object in the file system. 

An example use case would be data processing:

```

@task
def process_data(input: pd.DataFrame) -> pd.DataFrame
   # make some magic
```

Where DataFrame are saved between tasks as `csv`.

This is probably a stretch and maybe out of scope for this AIP (or require a new AIP). But wanted to put this out there.

### PR DESCRIPTION
**Pull request by turbaszek**
closes: #8059

---
Make sure to mark the boxes below before creating PR: [x]

- [ ] Description above provides context of the change
- [ ] Unit tests coverage for changes (not needed for documentation changes)
- [ ] Target Github ISSUE in description if exists
- [ ] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [ ] Relevant documentation is updated including usage instructions.
- [ ] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/config_templates/default_airflow.cfg [hunk 1]
```diff
@@ -208,6 +208,10 @@ max_num_rendered_ti_fields_per_task = 30
 # On each dagrun check against defined SLAs
 check_slas = True
 
+# Path to custom XCom class that will be used to store and resolve operators results
+# Example: xcom_class = path.to.CustomXCom
+xcom_class =
```
Comment (casassg) [inline airflow/config_templates/default_airflow.cfg]: Should we set the default here for visibility?

**Review by casassg — APPROVED** [review]
lgtm, small nitpick

**Comment by evgenyshulman** [comment]
are we sure that we want to implement XCom "override"?  This is relevant for XComAarg only, I was thinking about being able to override XComArg implementation, or control it if it's required. 
What if implementation still wants to use basic XCom.. 

Anyway, let's see how it looks when it combines with XArg implementation. Probably, is the only way to validate this implementation is to implement "working" customization that will use this feature end to end.

**Comment by turbaszek** [comment]
Will add @casassg suggestion and let's move to apache for further discussion

**Comment by turbaszek** [comment]
Moved to https://github.com/apache/airflow/pull/8560

**Pull request closed without merging**